File: app/src/main/java/com/ssy/ferry/trace/Tracer.java
package com.ssy.ferry.trace;

import com.ssy.ferry.listener.LooperObserver;

/**
 * 2019-10-21
 *
 * @author Mr.S
 */
public abstract class Tracer extends LooperObserver {

    public abstract void startTrace();
    public abstract void stopTrace();


}

File: app/src/main/java/com/ssy/ferry/mytrace/Tracer.kt
package com.ssy.ferry.mytrace

/**
 * 2019-10-16
 * @author Mr.S
 */
interface Tracer {
    /**
     * 开始
     */
    fun start()

    /**
     * 结束
     */
    fun end()

    /**
     * 是否存活
     */
    fun isAlive(): Boolean




}
